src/gtfs/read.rs

```diff
@@ -199,12 +220,186 @@ pub fn read_stops<P: AsRef<path::Path>>(
     (stopareas, stoppoints)
 }
 
+#[derive(Eq, PartialEq)]
+enum RouteReadType {
+    RouteAsNtmRoute,
+    RouteAsNtmLine,
+}
+
+fn define_route_file_read_mode(gtfs_routes: &Vec<Route>) -> RouteReadType {
+    for (i1, r1) in gtfs_routes.iter().enumerate() {
+        for (i2, r2) in gtfs_routes.iter().enumerate() {
+            if i2 <= i1 {
+                continue;
+            }
+            if r1.agency_id == r2.agency_id {
+                if r1.short_name != "" {
+                    if r1.short_name == r2.short_name {
+                        return RouteReadType::RouteAsNtmRoute;
+                    }
+                } else {
+                    if r1.long_name == r2.long_name {
+                        return RouteReadType::RouteAsNtmRoute;
+                    }
+                }
+            }
+        }
+    }
+    RouteReadType::RouteAsNtmLine
+}
+
+fn get_commercial_mode_label(route_type: u8) -> String {
+    let result = match route_type {
+        0 => "Tram, Streetcar, Light rail",
+        1 => "Subway, Metro",
+        2 => "Rail",
+        4 => "Ferry",
+        5 => "Cable car",
+        6 => "Gondola",
+        7 => "Funicular",
+        _ => "Bus",
+    };
+    result.to_string()
+}
+
+fn get_physical_mode(route_type: u8) -> objects::PhysicalMode {
+    let mode = match route_type {
+        0 => objects::PhysicalMode {
+            id: "Tramway".to_string(),
+            name: "Tramway".to_string(),
+            co2_emission: None,
+        },
+        1 => objects::PhysicalMode {
+            id: "Metro".to_string(),
+            name: "Métro".to_string(),
+            co2_emission: None,
+        },
+        2 => objects::PhysicalMode {
+            id: "Train".to_string(),
+            name: "Train".to_string(),
+            co2_emission: None,
+        },
+        4 => objects::PhysicalMode {
+            id: "Ferry".to_string(),
+            name: "Ferry".to_string(),
+            co2_emission: None,
+        },
+        5 => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        6 => objects::PhysicalMode {
+            id: "Boat".to_string(),
+            name: "Navette maritime/fluviale".to_string(),
+            co2_emission: None,
+        },
+        7 => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        _ => objects::PhysicalMode {
+            id: "Bus".to_string(),
+            name: "Bus".to_string(),
+            co2_emission: None,
+        },
+    };
+    mode
+}
+
+fn get_modes_from_gtfs(
+    gtfs_routes: &Vec<Route>,
```

Comment:
why?

Reply:
because all you can do with `&Vec<T>`, you can do it with `&[T]`, and there is cases you can use `&[T]` but not `&Vec<T>` (subslice, array on stack...).